modules/sublimeBidAdapter.js

```diff
@@ -1,135 +1,282 @@
-        bidRequestsCount: bid.bidRequestsCount,
-        requestId: bid.bidId,
-        sizes: bid.sizes.map(size => ({
-          w: size[0],
-          h: size[1],
-        })),
-        transactionId: bid.transactionId,
-        zoneId: bid.params.zoneId,
-      };
+/**
+ * Send pixel to our debug endpoint
+ * @param {string} eventName - Event name that will be send in the e= query string
+ * @param {Boolean} isMandatoryPixel - If set to true, will always send the pixel
+ */
+function sendEvent(eventName, isMandatoryPixel) {
+  let shoudSendPixel = (isMandatoryPixel || state.debug);
+  let eventObject = {
+    tse: Date.now(),
+    z: state.zoneId,
+    e: eventName,
+    src: 'pa',
+    trId: state.transactionId,
+    ver: SUBLIME_VERSION,
+  };
 
-      let protocol = bid.params.protocol || DEFAULT_PROTOCOL;
-      let bidHost = bid.params.bidHost || DEFAULT_BID_HOST;
-      let payload = Object.assign({}, commonPayload, bidPayload);
-
-      return {
-        method: 'POST',
-        url: protocol + '://' + bidHost + '/bid',
-        data: payload,
-        options: {
-          contentType: 'application/json',
-          withCredentials: true
-        },
-      };
+  if (shoudSendPixel) {
+    log('Sending pixel for event: ' + eventName, eventObject);
+
+    let queryString = url.formatQS(eventObject);
+    utils.triggerPixel('https://' + SUBLIME_ANTENNA + '/?' + queryString);
+  } else {
+    log('Not sending pixel for event (use debug: true to send it): ' + eventName, eventObject);
+  }
+}
+
+/**
+ * Function sendPixel, use it to send pixel to antenna
+ * @param {String} pixelName
+ */
+function sendPixel(pixelName) {
+  let pixelObject = {
+    t: Date.now(),
+    z: state.zoneId,
+    e: pixelName,
+    et: Math.round(window.performance.now()),
+  };
+
+  log('Sending pixel: ' + pixelName, pixelObject);
+
+  let queryString = url.formatQS(pixelObject);
+  utils.triggerPixel('https://' + SUBLIME_ANTENNA + '/?' + queryString);
+}
+
+/**
+ * Determines whether or not the given bid request is valid.
+ *
+ * @param {BidRequest} bid The bid params to validate.
+ * @return {Boolean} True if this is a valid bid, and false otherwise.
+ */
+function isBidRequestValid(bid) {
+  return !!Number(bid.params.zoneId);
+}
+
+/**
+ * Make a server request from the list of BidRequests.
+ *
+ * @param {BidRequest[]} validBidRequests An array of bids
+ * @param {Object} bidderRequest - Info describing the request to the server.
+ * @return ServerRequest Info describing the request to the server.
+ */
+function buildRequests(validBidRequests, bidderRequest) {
+  window.sublime = window.sublime || {};
+
+  let commonPayload = {
+    sublimeVersion: SUBLIME_VERSION,
+    // Current Prebid params
+    prebidVersion: '$prebid.version$',
+    currencyCode: config.getConfig('currency.adServerCurrency') || DEFAULT_CURRENCY,
+    timeout: typeof bidderRequest !== 'undefined' ? bidderRequest.timeout : config.getConfig('bidderTimeout'),
```

Comment:
Beware: typeof null is 'object' ...

At a minimum, I prefer:
`(typeof bidderRequest === 'object' && !!bidderRequest)`